pkg/auth/service.go

```diff
@@ -955,7 +955,7 @@ func (s *AuthService) GetCredentialsForUser(ctx context.Context, username, acces
 func (s *AuthService) GetCredentials(ctx context.Context, accessKeyID string) (*model.Credential, error) {
 	return s.cache.GetCredential(accessKeyID, func() (*model.Credential, error) {
 		m := &model.UserData{}
-		itr, err := s.store.Scan(ctx, m.ProtoReflect().Type(), model.PartitionKey, model.UserPath(""), []byte(""))
+		itr, err := kv.NewPrimaryIterator(ctx, s.store, m.ProtoReflect().Type(), model.PartitionKey, model.UserPath(""), kv.IteratorOptionsAfter([]byte("")))
```

Comment:
Why not use Scan? Do we not need Scan in the interface anymore?

Reply:
We have 'Scan' on the kv level which doesn't support messages.
The 'Scan' we used to call on the message store was just returning a new iterator as we do here.